libcaf_net/caf/net/http/header.hpp

```diff
@@ -0,0 +1,154 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#pragma once
+
+#include "caf/net/fwd.hpp"
+
+#include "caf/config_value.hpp"
+#include "caf/detail/net_export.hpp"
+#include "caf/string_algorithms.hpp"
+
+#include <string_view>
+#include <vector>
+
+namespace caf::net::http {
+
+/// Encapsulates meta data for HTTP header fields. This class represents a base
+/// class used for HTTP request and response representations, each providing
+/// additional message specific methods.
+class CAF_NET_EXPORT header {
+public:
+  /// Virtual destructor.
+  virtual ~header() = default;
```

Comment:
This will create a weak vtable. Please define out-of-line in the .cpp file.

```suggestion
  virtual ~header();
```